syntax = "proto3";

message Empty {}

// Define a message for chat messages
message ChatMessage {
    string message_id = 1;
    string from_user_id = 2;
    string to_user_id = 3;
    string timestamp = 4;
    string content_type = 5;
    string content = 6;
    string content_url = 7;
    string caption = 8;
    string hotel_id = 9; // Add this field to specify the hotel's ID


}

service ChatServer {
    rpc ChatStream (Empty) returns (stream ChatMessage);
    rpc SendChatMessage (ChatMessage) returns (Empty);
}


